pkg/internal/token/ropc.go

```diff
@@ -41,11 +55,57 @@ func newResourceOwnerToken(oAuthConfig adal.OAuthConfig, clientID, username, pas
 		resourceID:  resourceID,
 		tenantID:    tenantID,
 		oAuthConfig: oAuthConfig,
+		popClaims:   popClaims,
 	}, nil
 }
 
+// Token fetches an azcore.AccessToken from the Azure SDK and converts it to an adal.Token for use with kubelogin.
 func (p *resourceOwnerToken) Token(ctx context.Context) (adal.Token, error) {
+	return p.TokenWithOptions(ctx, nil)
+}
+
+func (p *resourceOwnerToken) TokenWithOptions(ctx context.Context, options *azcore.ClientOptions) (adal.Token, error) {
 	emptyToken := adal.Token{}
+	authorityFromConfig := p.oAuthConfig.AuthorityEndpoint
+	clientOpts := azcore.ClientOptions{Cloud: cloud.Configuration{
+		ActiveDirectoryAuthorityHost: authorityFromConfig.String(),
+	}}
+	if options != nil {
+		clientOpts = *options
+	}
+	var err error
+	scopes := []string{p.resourceID + defaultScope}
+	if len(p.popClaims) > 0 {
+		// If PoP token support is enabled and the correct u-claim is provided, use the MSAL
+		// token provider to acquire a new token
+		token, expirationTimeUnix, err := pop.AcquirePoPTokenByUsernamePassword(
+			ctx,
+			p.popClaims,
+			scopes,
+			authorityFromConfig.String(),
+			p.clientID,
+			p.username,
+			p.password,
+			&clientOpts,
+		)
+		if err != nil {
+			return emptyToken, fmt.Errorf("failed to create PoP token using resource owner flow: %w", err)
+		}
+
+		// azurecore.AccessTokens have ExpiresOn as Time.Time. We need to convert it to JSON.Number
+		// by fetching the time in seconds since the Unix epoch via Unix() and then converting to a
+		// JSON.Number via formatting as a string using a base-10 int64 conversion.
+		expiresOn := json.Number(strconv.FormatInt(expirationTimeUnix, 10))
+
+		// Re-wrap the azurecore.AccessToken into an adal.Token
+		return adal.Token{
+			AccessToken: token,
+			ExpiresOn:   expiresOn,
+			Resource:    p.resourceID,
+		}, nil
+	}
+
+	// otherwise, if PoP token flow is not enabled, use the default flow
```

Comment:
can we always use msal? it can hit two birds (enabling pop and deprecating adal)in one stone :)

Reply:
Discussed offline, since a lot of other flows are still using `adal`, created a follow-up WI to migrate everything to MSAL: https://msazure.visualstudio.com/AzureArcPlatform/_workitems/edit/26914387